src/content/language/macros.md

```diff
@@ -0,0 +1,287 @@
+---
+title: Macros (experimental)
+description: Learn about the experimental macros feature as it develops.
+---
+
+[The Dart macro system][spec] is a major new language feature
+***currently under development*** which adds support for
+[static meta-programming][motivation] to the Dart language.
+
+A Dart macro is a user-defineable piece of code that takes in other code as parameters
+and operates on it during compile time to create, modify, or add declarations.
```

Comment:
I prefer something like "real-time" or "continously as you edit" here rather than "compile time". While it's technically correctd, many perceive "compile time" as "when you run the compiler/invoke run", and not as an integrated thing that happens continously.